.github/workflows/tox.yml

```diff
@@ -34,6 +34,12 @@ jobs:
     - name: Install python dependencies
       run: |
         python -m pip install --upgrade pip setuptools wheel
+    - name: Set up C++ devtools
+      uses: seanmiddleditch/gha-setup-ninja@v3
```

Comment:
If you want to keep it, why this external ghaction, instead of just `sudo apt install ninja-build` below?

Reply:
This will work on Mac OS on the CI. I didn't think too much about the library dependencies so I just used apt, those are hacky anyways until they are properly done with spack or something,

Reply:
Changed it to use ninja and cmake from PyPI.